### DISCUSSION TITLE
Implementation of new Models/Plugins

### DISCUSSION THREAD

**Discussion by JamesonBR** [exadel-inc/CompreFace#1104] [Q&A]
Hello, I would like to know where the models are saved and how to implement a new model in the API.

**Comment by pospielov** [comment]
Models are stored in google drive, you can find the links in this files:
https://github.com/exadel-inc/CompreFace/blob/master/embedding-calculator/src/services/facescan/plugins/insightface/insightface.py#L136
https://github.com/exadel-inc/CompreFace/blob/master/embedding-calculator/src/services/facescan/plugins/facenet/facenet.py#L141
To add plugin, you need to add custom implementation, look how it was done for gender and age plugins:
https://github.com/exadel-inc/CompreFace/blob/master/embedding-calculator/src/services/facescan/plugins/agegender/agegender.py
